### ISSUE TITLE
Review chained protected access in test_learning_rate_scheduler

### ISSUE THREAD

**Issue by rg936672** [gchq/Vanguard#207]
### What's the issue?

In `test_learning_rate_scheduler.BasicTests.test_learning_rate_is_stepped`, we have the line
```python
current_lr = self.controller._smart_optimiser._internal_optimiser._applied_scheduler.get_lr()[0]
```
This reads as a _bit_ of a code smell to me - perhaps we should be exposing the current learning rate as a read-only property in each of these internal classes? If not, simply remove the associated TODO and close this issue.

**Comment by tm91236** [comment]
Vanguard is being archived
